src/orion/core/worker/experiment.py

```diff
@@ -163,12 +165,19 @@ def __init__(
         self.metadata = metadata or {}
         self.max_trials = max_trials
         self.max_broken = max_broken
-
         self.algorithms = algorithms
         self.working_dir = working_dir
 
-        self._storage = get_storage()
-        self._node = ExperimentNode(self.name, self.version, experiment=self)
+        self._storage = storage or setup_storage()
```

Comment:
The `or` should never be executed here now, otherwise the assert above would raise.

Reply:
done